account_invoice_split/models/account_invoice.py

```diff
@@ -0,0 +1,66 @@
+from odoo import fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoiceInherit(models.Model):
+    _inherit = "account.invoice"
+
+    split_id = fields.Many2one(
+        string="Split From",
+        comodel_name="account.invoice",
+        help="INV Split From Ref:",
+    )
+
+    def btn_split_quotation(self):
+        """
+        Define function to split invoice when we click on button
+        :return: New INV view
+        """
+        self.ensure_one()
+
+        # split invoice
+        lines_to_split = [line for line in self.invoice_line_ids if line.split]
```

Comment:
I'm wondering how these lines will get their `split=True` value. I don't see anything that sets it - but maybe I'm overlooking.

Reply:
It's a field on the line that is displayed on the invoice form :-)

Reply:
```xml
            <xpath expr="//field[@name='invoice_line_ids']//tree/field[last()]"
                   position="before">
                <field name="split"
                       groups="account.group_show_checkbox_button"/>
            </xpath>
```